cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/HashLoginServiceApiCredentials.java

```diff
@@ -0,0 +1,333 @@
+        this.ownerReference = ownerReference;
+        this.ccSpec = ccSpec;
+
+        if (ccSpec.getApiUsers() != null) {
+            if (ccSpec.getApiUsers() instanceof HashLoginServiceApiUsers apiUsers) {
+                validateHashLoginServiceApiUsersConfiguration(apiUsers);
+                this.userManagedApiSecretName = apiUsers.getValueFrom().getSecretKeyRef().getName();
+                this.userManagedApiSecretKey = apiUsers.getValueFrom().getSecretKeyRef().getKey();
+            } else {
+                throw new InvalidResourceException("Unsupported API user configuration type " + ccSpec.getApiUsers().getType());
+            }
+        } else {
+            this.userManagedApiSecretName = null;
+            this.userManagedApiSecretKey = null;
+        }
+    }
+
+    /**
+     * @return Returns user-managed API credentials secret name
+     */
+    public String getUserManagedApiSecretName() {
+        return this.userManagedApiSecretName;
+    }
+
+    /**
+     * @return Returns user-managed API credentials secret key
+     */
+    /* test */ String getUserManagedApiSecretKey() {
+        return this.userManagedApiSecretKey;
+    }
+
+    /**
+     * Checks if Cruise Control spec has valid ApiUsers config.
+     *
+     * @param apiUsers The API users configuration
+     */
+    private static void validateHashLoginServiceApiUsersConfiguration(HashLoginServiceApiUsers apiUsers)  {
+        if (apiUsers.getValueFrom() == null
+                || apiUsers.getValueFrom().getSecretKeyRef() == null
+                || apiUsers.getValueFrom().getSecretKeyRef().getName() == null
+                || apiUsers.getValueFrom().getSecretKeyRef().getKey() == null) {
+            throw new InvalidResourceException("The configuration of the Cruise Control REST API users " +
+                    "referenced in spec.cruiseControl.apiUsers is invalid.");
+        }
+    }
+
+    /**
+     * Parse String containing API credential config into map of API user entries.
+     *
+     * @param config API credential config file as a String.
+     *
+     * @return map of API credential entries containing username, password, and role.
+     */
+    /* test */ static Map<String, UserEntry> parseEntriesFromString(String config) {
+        Map<String, UserEntry> entries = new HashMap<>();
+
+        if (config.isEmpty()) {
+            return entries;
+        }
+
+        for (String line : config.split("\n")) {
+            Matcher matcher = HASH_LOGIN_SERVICE_PATTERN.matcher(line);
+            if (matcher.matches()) {
+                String[] parts = line.replaceAll("\\s", "").split(":");
+                String username = parts[0];
+                String password = parts[1].split(",")[0];
+                Role role = Role.fromString(parts[1].split(",")[1]);
+                if (entries.containsKey(username)) {
+                    throw new InvalidConfigurationException("Duplicate username found: " + "\"" + username + "\". "
```

Comment:
Why are you everywhere doing these unnecessary splits in the strings?

```suggestion
                    throw new InvalidConfigurationException("Duplicate username found: \"" + username + "\". "
```